walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,81 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    private void addTimeRange(List<TimeRange> timeList, int startTime, int endTime, boolean includeEnd) {
+        timeList.add((TimeRange.fromStartEnd(  
+            startTime,
+            endTime,
+            includeEnd)));
+    }
+
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        List<TimeRange> eventTimeRanges = new ArrayList<>();
+        List<TimeRange> openTimeRanges = new ArrayList<>();
+
+        if(request.getAttendees() != null && !request.getAttendees().isEmpty()) {
```

Comment:
This whole function is deeply nested; consider restructuring to avoid this. Deeply nested functions are difficult to follow and keep track of what level of control block each segment of code is at. 
Some easy ways to get around this: 
1) guard statements. Currently, you start off with an if statement that ensures you have valid attendees, then embeds all your logic in the body of the if statement. You can instead use an if statement to check for the /invalid/ condition, and return early. Then your main logic continues at the default identation level. 
2) where appropriate,  you can pull logic out into small function calls.